src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs

```diff
@@ -114,201 +165,195 @@ public string CreateSignedHttpRequest(SignedHttpRequestDescriptor signedHttpRequ
         {
-            if (payload == null)
-                throw LogHelper.LogArgumentNullException(nameof(payload));
-
-            var httpRequestUri = signedHttpRequestDescriptor.HttpRequestData.Uri;
-
-            if (httpRequestUri == null)
+            if (signedHttpRequestDescriptor.HttpRequestData.Uri == null)
                 throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestDescriptor.HttpRequestData.Uri));
 
-            if (!httpRequestUri.IsAbsoluteUri)
-                throw LogHelper.LogExceptionMessage(new SignedHttpRequestCreationException(LogHelper.FormatInvariant(LogMessages.IDX23001, httpRequestUri.OriginalString)));
+            if (!signedHttpRequestDescriptor.HttpRequestData.Uri.IsAbsoluteUri)
+                throw LogHelper.LogExceptionMessage(new SignedHttpRequestCreationException(LogHelper.FormatInvariant(LogMessages.IDX23001, signedHttpRequestDescriptor.HttpRequestData.Uri.OriginalString)));
 
             // https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#section-3
             // u claim: The HTTP URL host component as a JSON string. This MAY include the port separated from the host by a colon in host:port format.
             // Including the port if it not the default port for the httpRequestUri scheme.
-            var httpUrlHostComponent = httpRequestUri.Host;
-            if (!httpRequestUri.IsDefaultPort)
-                httpUrlHostComponent = $"{httpUrlHostComponent}:{httpRequestUri.Port}";
+            string httpUrlHostComponent = signedHttpRequestDescriptor.HttpRequestData.Uri.Host;
+            if (!signedHttpRequestDescriptor.HttpRequestData.Uri.IsDefaultPort)
+                httpUrlHostComponent = $"{httpUrlHostComponent}:{signedHttpRequestDescriptor.HttpRequestData.Uri.Port}";
 
-            payload.Add(SignedHttpRequestClaimTypes.U, httpUrlHostComponent);
+            writer.WriteString(SignedHttpRequestClaimTypes.U, httpUrlHostComponent);
         }
 
         /// <summary>
-        /// Adds the 'm' claim to the <paramref name="payload"/>.
+        /// Adds the 'm' claim to the <paramref name="writer"/>.
         /// </summary>
-        /// <param name="payload">HttpRequest payload represented as a <see cref="Dictionary{TKey, TValue}"/>.</param>
+        /// <param name="writer"><see cref="Utf8JsonWriter"/></param>
         /// <param name="signedHttpRequestDescriptor">A structure that wraps parameters needed for SignedHttpRequest creation.</param>
         /// <remarks>
         /// This method will be executed only if <see cref="SignedHttpRequestCreationParameters.CreateP"/> is set to <c>true</c>.
         /// </remarks>  
-        internal virtual void AddPClaim(Dictionary<string, object> payload, SignedHttpRequestDescriptor signedHttpRequestDescriptor)
+        internal virtual void AddPClaim(ref Utf8JsonWriter writer, SignedHttpRequestDescriptor signedHttpRequestDescriptor)
```

Comment:
>AddPClaim( [](http://example.com/codeflow?start=30&length=10)

comment says to add 'm' claim but it's called `AddPClaim`?

Reply:
@jennyf19 addressed in d789adbb30b8093a0eff70ee545983195935366f